pkg/wwan/usr/bin/wwan-qmi.sh

```diff
@@ -136,85 +200,115 @@ get_get_sim_imsi() {
 
 qmi_get_sim_cards() {
   # FIXME XXX Limited to a single SIM card
-  if ! ICCID="$(get_get_sim_iccid)"; then
+  if ! ICCID="$(qmi_get_sim_iccid)"; then
     echo "[]"
     return 1
   fi
-  if ! IMSI="$(get_get_sim_imsi)"; then
+  if ! IMSI="$(qmi_get_sim_imsi)"; then
     echo "[]"
     return 1
   fi
   SIM="$(json_struct "$(json_str_attr "iccid" "$ICCID")" "$(json_str_attr "imsi" "$IMSI")")\n"
   printf "%b" "$SIM" | json_array
 }
 
+qmi_get_ip_settings() {
+  if ! SETTINGS="$(qmi --wds-get-current-settings)"; then
+    return 1
+  fi
+  IP=$(parse_modem_attr "$SETTINGS" "IPv4 address")
+  SUBNET=$(parse_modem_attr "$SETTINGS" "IPv4 subnet mask")
+  GW=$(parse_modem_attr "$SETTINGS" "IPv4 gateway address")
+  DNS1=$(parse_modem_attr "$SETTINGS" "IPv4 primary DNS")
+  DNS2=$(parse_modem_attr "$SETTINGS" "IPv4 secondary DNS")
```

Comment:
The old code looked for dns1 (and dns2) in the json output. Why not use the json format here?

Reply:
Does qmi not have such an output format?

Reply:
Sadly qmicli does not allow to switch to json output.
uqmi is perhaps more scripting friendly in this regard, but then it does not support many QMI features, including location tracking, so we have to opt for qmicli.